### ISSUE TITLE
Non 'AssertionError' with Async Mocha leads to timeout

### ISSUE THREAD

**Issue by dvlsg** [sindresorhus/gulp-mocha#68]
I noticed a potential issue while using gulp-mocha with asynchronous mocha tests and an assertion library which did not throw an error with error.name equal to "AssertionError". Essentially, if an AssertionError is not thrown, then gulp-mocha will hang on that error until mocha's internal timeout is hit instead of continuing on to the next test.

I noticed that if I changed [Line 34-39 of index.js](https://github.com/sindresorhus/gulp-mocha/blob/master/index.js#L34-L39) to be this, the timeout problem was resolved:
```
if (runner && runner.uncaught) {
  runner.uncaught(err);
} else {
  clearCache();
  stream.emit('error', new gutil.PluginError('gulp-mocha', err));
}
```

I would worry that this might break some intended functionality, however, especially with users that would be expecting the error to be emitted by the stream.

Here is a link to a [gist](https://gist.github.com/dvlsg/eb427a69473d50cc80d6) of the gulpfile and the test-simple file I was using to reproduce the issue. I also tinkered with catching the error from pipe.on('error', callback), but didn't come up with any results with that either.

**Comment by seanzer** [comment]
+1

**Comment by Jameskmonger** [comment]
This has been fixed by #114.
